pkg/microservice/packager/executor/executor.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2021 The KodeRover Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executor
+
+import (
+	commonconfig "github.com/koderover/zadig/pkg/config"
+	"github.com/koderover/zadig/pkg/microservice/packager/core/service"
+	"github.com/koderover/zadig/pkg/setting"
+	"github.com/koderover/zadig/pkg/tool/log"
+)
+
+func Execute() error {
+	log.Init(&log.Config{
+		Level:       commonconfig.LogLevel(),
+		NoCaller:    true,
+		NoLogLevel:  true,
+		Development: commonconfig.Mode() != setting.ReleaseMode,
+	})
+
+	pred, err := service.NewPackager()
+	if err != nil {
+		log.Errorf("Failed to start packager, error: %s", err)
+		return err
+	}
+
+	if err := pred.BeforeExec(); err != nil {
+		log.Errorf("Failed to run before exec step, error: %s", err)
+		return err
+	}
+
+	if err := pred.Exec(); err != nil {
```

Comment:
What's the reson that the executor should be aware of the steps of Packager do? Can we simply expose a "Exec" method?

Reply:
Referring to to code of reaper/predator, doing some prepare preparation in before exec. I will fix it since it's not necessary indeed.